libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -304,7 +336,9 @@ Stack StackLayoutGenerator::propagateStackThroughOperation(Stack _exitStack, CFG
 			stack.pop_back();
 		else if (auto offset = util::findOffset(stack | ranges::views::reverse | ranges::views::drop(1), stack.back()))
 		{
-			if (*offset + 2 < 16)
+			// *offset == n requires, after popping the topmost slot, a DUP<n+1> to recover it.
+			// So if *offset + 1 is within stack reach, we can pop.
+			if (*offset + 1 <= m_reachableStackDepth)
```

Comment:
This is a bit funky on develop already - I'll also look at this separately in https://github.com/ethereum/solidity/tree/stackShufflingDupLimit - but it seems to have some odd effects. Without it it's trickier to produce a ``dupn{256}`` test, though.